src/CalculatorUI/Common/AppLifecycleLogger.cs

```diff
@@ -0,0 +1,167 @@
+            PopulateAppInfo(fields);
+            LogAppLifecycleEvent("ModernAppResume_VisibleComplete", fields);
+
+        }
+
+        public void ResizeUIResponsive()
+        {
+            ResizeUIResponsive(ApplicationView.GetForCurrentView().Id);
+        }
+
+        public void ResizeVisibleComplete()
+        {
+            ResizeVisibleComplete(ApplicationView.GetForCurrentView().Id);
+        }
+
+        public void ResizeUIResponsive(int viewId)
+        {
+            if (!GetTraceLoggingProviderEnabled())
+                return;
+
+            LoggingFields fields = new LoggingFields();
+            PopulateAppInfo(fields);
+            fields.AddInt32("ViewId", viewId);
+            LogAppLifecycleEvent("ModernAppResize_UIResponsive", fields);
+        }
+
+        public void ResizeVisibleComplete(int viewId)
+        {
+            if (!GetTraceLoggingProviderEnabled())
+                return;
+
+            LoggingFields fields = new LoggingFields();
+            PopulateAppInfo(fields);
+            fields.AddInt32("ViewId", viewId);
+            LogAppLifecycleEvent("ModernAppResize_VisibleComplete", fields);
+        }
+
+        // Make the object construction private to allow singleton access to this class
+        private AppLifecycleLogger()
+        {
+            m_appLifecycleProvider = new LoggingChannel(
+                        "Microsoft.Windows.AppLifeCycle",
+                        new LoggingChannelOptions(new Guid(0x4f50731a, 0x89cf, 0x4782, 0xb3, 0xe0, 0xdc, 0xe8, 0xc9, 0x4, 0x76, 0xba)),
+                        new Guid(0xef00584a, 0x2655, 0x462c, 0xbc, 0x24, 0xe7, 0xde, 0x63, 0xe, 0x7f, 0xbf));
+        }
+
+        // Any new Log method should
+        // a) Decide the level of logging. This will help us in limiting recording of events only up to a certain level. See this link for guidance
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/aa363742(v=vs.85).aspx We're using Verbose level for events that are called frequently and
+        // needed only for debugging or capturing perf for specific scenarios b) Should decide whether or not to log to diagnostics and pass
+        // TraceLoggingKeyword(MICROSOFT_KEYWORD_LEVEL_3) accordingly c) Should accept a variable number of additional data arguments if needed
+        private void LogAppLifecycleEvent(string eventName, LoggingFields fields)
+        {
+            m_appLifecycleProvider.LogEvent(
+                eventName, fields, LoggingLevel.Information, new LoggingOptions(Globals.MICROSOFT_KEYWORD_LEVEL_3 | WinMeta.WINEVENT_KEYWORD_RESPONSE_TIME));
+        }
+
+        private void PopulateAppInfo(LoggingFields fields)
+        {
+
+            var appId = CoreApplication.Id;
+            var aumId = Package.Current.Id.FamilyName + "!" + appId;
+            var packageFullName = Package.Current.Id.FullName;
+            var psmKey = Package.Current.Id.FullName + "+" + appId;
+
+            fields.AddString("AumId", aumId);
```

Comment:
Is "AumId" a key? if it is, it's better to set these keys as constant instead of hard code.

Reply:
same as above